packages/js-sync-with-portabl/src/index.ts

```diff
@@ -1,182 +1,144 @@
-import { Datapoints, Options } from './lib/types';
+import {
+  IframeWidgetClientMesssageType,
+  IncomingMessageDataType,
+  Options,
+} from './lib/types';
 import { withRetries, MAX_RETRIES } from './lib/withRetries';
 import {
-  PREREQS_FAILED,
-  PREREQS_SUCCESS,
-  SYNC_ACCEPT_SUCCESS,
-  SYNC_ACK,
-  SYNC_PASSPORT_CLOSED,
-  SYNC_PASSPORT_READY,
-  SYNC_READY,
-  SYNC_REQUEST_ACK,
-  SYNC_REQUEST_ERROR,
-  USER_AUTHENTICATED,
-  WIDGET_READY,
+  OutgoingPostMessageEvent,
+  IncomingPostMessageEvent,
+  DEFAULT_ROOT_SELECTOR,
 } from './lib/constants';
 import {
   createContainer,
   createModal,
-  createIframe,
+  createIframeModal,
   createIframeWidget,
 } from './lib/syncElements';
+import { createPostMessageSenderClient } from './lib/createPostMessageSendClient';
+
+// Define message handler outside of create function
+// so that it can be referenced for removal of the previously
+// attached event listener when reinitializing
+let messageHandler: (e: MessageEvent<IncomingMessageDataType>) => Promise<void>;
 
 export async function createSyncWithPortabl(options: Options): Promise<void> {
   const {
     widgetBaseUrl = 'https://widgets.getportabl.com',
-    getPrereqs,
-    onUserConsent,
-    rootSelector,
+    getSyncContext,
+    prepareSync,
+    rootSelector = DEFAULT_ROOT_SELECTOR,
     // providerName,
```

Comment:
is `providerName` still needed?

Reply:
Yup added it back - we removed it temporarily.